server/models/Attachment.js

```diff
@@ -51,10 +50,6 @@ const Attachment = sequelize.define(
   }
 );
 
-Attachment.beforeDestroy(async (model) => {
```

Comment:
This was being triggered when the document was placed in the trash, resulting in the file being immediately deleted from s3. You can reproduce this in production today by createing a doc with an image, putting it in the trash and reloading moments later – the image is gone and there's no way to recover.